Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift

```diff
@@ -19,243 +19,196 @@ struct RestoreWalletContainerView: View {
-                withAnimation {
-                  showingRecoveryPhase.toggle()
-                }
-              }
-          }
-          .accessibilityElement(children: .combine)
-          .frame(maxWidth: .infinity, alignment: .leading)
-        }
-        HStack {
-          HStack {
-            Toggle(Strings.Wallet.restoreWalletShowRecoveryPhrase, isOn: $showingRecoveryPhase)
-              .labelsHidden()
-              .toggleStyle(SwitchToggleStyle(tint: .accentColor))
-              .scaleEffect(0.75)
-              .padding(-6)
-            Text(Strings.Wallet.restoreWalletShowRecoveryPhrase)
-              .font(.footnote)
-              .onTapGesture {
-                withAnimation {
-                  showingRecoveryPhase.toggle()
-                }
+      let columns: [GridItem] = (0..<numberOfColumns).map { _ in .init(.flexible()) }
+      ScrollView {
+        LazyVGrid(columns: columns, spacing: 8) {
+          ForEach(self.recoveryWords.indices, id: \.self) { index in
+            VStack(alignment: .leading, spacing: 10) {
+              if isRevealRecoveryWords {
+                TextField(String.localizedStringWithFormat(Strings.Wallet.restoreWalletPhrasePlaceholder, (index + 1)), text: $recoveryWords[index])
+                  .autocapitalization(.none)
+                  .disableAutocorrection(true)
+                  .foregroundColor(Color(.braveLabel))
+              } else {
+                SecureField(String.localizedStringWithFormat(Strings.Wallet.restoreWalletPhrasePlaceholder, (index + 1)), text: $recoveryWords[index])
+                  .textContentType(.newPassword)
               }
-          }
-          .accessibilityElement(children: .combine)
-          .accessibilityAddTraits(.isButton)
-          Spacer()
-          Button(action: {
-            if let string = UIPasteboard.general.string {
-              phrase = string
+              Divider()
             }
-          }) {
-            Label(Strings.Wallet.pasteFromPasteboard, braveSystemImage: "leo.copy.plain-text")
-              .labelStyle(.iconOnly)
           }
         }
       }
-      VStack {
-        Text(Strings.Wallet.restoreWalletNewPasswordTitle)
-          .font(.subheadline.weight(.medium))
-        SecureField(Strings.Wallet.passwordPlaceholder, text: $password)
-          .textContentType(.newPassword)
-          .textFieldStyle(BraveValidatedTextFieldStyle(error: restoreError, when: .requirementsNotMet))
-        SecureField(Strings.Wallet.repeatedPasswordPlaceholder, text: $repeatedPassword)
-          .textContentType(.newPassword)
-          .textFieldStyle(BraveValidatedTextFieldStyle(error: restoreError, when: .inputsDontMatch))
+      .frame(height: staticGridsViewHeight)
+      .padding(.horizontal)
+      if isShowingPhraseError {
+        errorLabel
+      }
+      HStack {
+        Spacer()
+        Button {
+          recoveryWords = isLegacyWallet ? .init(repeating: "", count: 12) : .init(repeating: "", count: 24)
```

Comment:
this is actually correct since `isLegacyWallet` meaning the user currently is inputting 24 recovery phrases and this button is to change to the not legacy (regular) wallet which only has 12 recovery phrases. i will add comment to eliminate the confusion.

Reply:
First comment still an issue (helps with line length) but second comment resolved nicely, thanks 👍

Reply:
updated in [1d4972d](https://github.com/brave/brave-ios/pull/7746/commits/1d4972df480781527014d1a81b091713712cdd90)